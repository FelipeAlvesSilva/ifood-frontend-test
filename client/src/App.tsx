import React from 'react';

function App() {
  return (
   <div>
     <h2>Spotifood</h2>
   </div>
  );
}

export default App;
