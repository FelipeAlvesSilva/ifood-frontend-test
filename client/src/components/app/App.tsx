import React from 'react';
import { withTheme } from 'styled-components';
import { TiWeatherPartlySunny as Sun } from 'react-icons/ti';
import { GiMoon as Moon } from 'react-icons/gi';
import { Switch } from 'antd';
import { appTheme } from '../theme/ThemeContext';

function App() {
  const themeToggle = appTheme();

  return (
    <div>
      <Switch
        onClick={() => themeToggle.toggle()}
        checkedChildren={<Moon />}
        unCheckedChildren={<Sun />}
        defaultChecked
      />
    </div>
  );
}

// HOC to obtain theme prop
export default withTheme(App);
