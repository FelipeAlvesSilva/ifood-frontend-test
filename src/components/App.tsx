import React, { Fragment } from 'react';
import { BrowserRouter as Router, Switch, Route, Redirect } from 'react-router-dom';
import Home from './home/Home';
import Logon from './logon/Logon';
import Fail from './fail/Fail';

const App = () => {
  return (
    <Fragment>
      <Router>
        <Switch>
          <Route path="/" exact component={Logon} />
          <Route path="/home" exact component={Home} />
          <Route path="/fail" exact component={Fail} />
          <Redirect from="*" to="/fail" />
        </Switch>
      </Router>
    </Fragment>
  )
};

export default App;