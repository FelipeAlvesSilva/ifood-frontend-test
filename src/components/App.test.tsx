import React from 'react';
import { render } from '@testing-library/react';
import App from '../components/App';

describe('Tests of App Component', () => {
  test('Component should not regress', () => {
    const componentContainer = render(<App />);
    expect(componentContainer).toMatchSnapshot();
  })
});

/**
 * Keep in mind:
  
  - Do the tests I am proposing to write help prevent regression?
  - Do these tests test how a user will interact with this feature?
  - If you find it difficult to test your React components they should probably be rewritten.
 * 
 */