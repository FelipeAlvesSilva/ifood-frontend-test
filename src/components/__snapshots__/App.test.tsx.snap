// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests of App Component Component should not regress 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        style="margin-left: auto; margin-right: auto; max-width: 30rem; text-align: center; padding: 4rem 0rem;"
      >
        <img
          alt="Spotify brand logo"
          class="sc-AxhCb dGMTeP"
          src="Spotify_Icon_CMYK_Black.png"
        />
        <h2
          class="display-4"
        >
          Welcome to Spotifood
        </h2>
        <p
          class="lead"
        >
          Here you can find all the most played playlists of the moment according to your profile 
          <span
            aria-label="Headhphone emoji"
            class="bounce"
            role="img"
          >
            ðŸŽ§
          </span>
        </p>
        <button
          class="ant-btn sc-AxiKw gwxEuR ant-btn-block"
          type="button"
        >
          <span>
            Login on Spotify
          </span>
        </button>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      style="margin-left: auto; margin-right: auto; max-width: 30rem; text-align: center; padding: 4rem 0rem;"
    >
      <img
        alt="Spotify brand logo"
        class="sc-AxhCb dGMTeP"
        src="Spotify_Icon_CMYK_Black.png"
      />
      <h2
        class="display-4"
      >
        Welcome to Spotifood
      </h2>
      <p
        class="lead"
      >
        Here you can find all the most played playlists of the moment according to your profile 
        <span
          aria-label="Headhphone emoji"
          class="bounce"
          role="img"
        >
          ðŸŽ§
        </span>
      </p>
      <button
        class="ant-btn sc-AxiKw gwxEuR ant-btn-block"
        type="button"
      >
        <span>
          Login on Spotify
        </span>
      </button>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
